python/lsst/resources/http.py

```diff
@@ -235,38 +265,88 @@ def _is_webdav_endpoint(path: ResourcePath | str) -> bool:
         True if the endpoint implements WebDAV, False if it doesn't.
     """
     log.debug("Detecting HTTP endpoint type for '%s'...", path)
+
+    # Send an OPTIONS request and inspect its response. An OPTIONS
+    # request does not need authentication of the client, so we don't need
+    # to provide a client certificate or a bearer token. We set a
+    # relatively short timeout since an OPTIONS request is relatively cheap
+    # for the server to compute.
+
+    # Create a session for configuring retries
+    retries = Retry(
+        # Total number of retries to allow. Takes precedence over other
+        # counts.
+        total=6,
+        # How many connection-related errors to retry on.
+        connect=3,
+        # How many times to retry on read errors.
+        read=3,
+        # How many times to retry on bad status codes.
+        status=5,
+        # Set of uppercased HTTP method verbs that we should retry on.
+        allowed_methods=frozenset(
+            [
+                "OPTIONS",
+            ]
+        ),
+        # HTTP status codes that we should force a retry on.
+        status_forcelist=frozenset(
+            [
+                requests.codes.too_many_requests,  # 429
+                requests.codes.internal_server_error,  # 500
+                requests.codes.bad_gateway,  # 502
+                requests.codes.service_unavailable,  # 503
+                requests.codes.gateway_timeout,  # 504
+            ]
+        ),
+        # Whether to respect 'Retry-After' header on status codes defined
+        # above.
+        respect_retry_after_header=True,
+    )
+
     try:
-        ca_cert_bundle = os.getenv("LSST_HTTP_CACERT_BUNDLE")
-        verify: bool | str = ca_cert_bundle if ca_cert_bundle else True
-        resp = requests.options(str(path), verify=verify, stream=False)
-        if resp.status_code not in (requests.codes.ok, requests.codes.created):
-            raise ValueError(
-                f"Unexpected response to OPTIONS request for {path}, status: {resp.status_code} {resp.reason}"
+        session = requests.Session()
+        session.mount(str(path), HTTPAdapter(max_retries=retries))
+        session.verify = os.environ.get("LSST_HTTP_CACERT_BUNDLE", True)
+        with session:
+            resp = session.options(
+                str(path),
+                stream=False,
+                timeout=(
+                    _timeout_from_environment("LSST_HTTP_TIMEOUT_CONNECT", 30.0),
```

Comment:
Should we be reading the environment variables here if we are saying that OPTIONS should use a shorter timeout than other activities?

Reply:
I thought using the timeout values specified by the user via the environment variables for `OPTIONS` as well would be better for consistency. In addition, using the environment allows the user to configure the environment for using a potential slow server, as opposed to hardcoding a magic value.

Reply:
Yes, okay. I was concerned that people might want different values for OPTIONS and other things but if the env var is only used for testing it doesn't matter.